devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy

```diff
@@ -12,98 +12,107 @@
+			state "default", label:'reset kWh', action:"reset"
+		}
+		standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
 
-        main (["power", "energy"])
-        details(["power", "energy", "reset", "refresh"])
-    }
+		main (["power", "energy"])
+		details(["power", "energy", "reset", "refresh"])
+	}
 }
 
+def getATTRIBUTE_READING_INFO_SET() { 0x0000 }
+def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
+
 def parse(String description) {
-    log.debug "description is $description"
-    def event = zigbee.getEvent(description)
-    if (event) {
-        log.info event
-        sendEvent(event)
-    } else {
-        List result = []
-        def descMap = zigbee.parseDescriptionAsMap(description)
-        log.debug "Desc Map: $descMap"
-                
-        List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
-        descMap.additionalAttrs.each {
-            attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)
+	if (event) {
+		log.info "event enter:$event"
+        if(event.name=="power"){
+            event.value =event.value/1000
+        }else if(event.name=="energy"){
+            event.value =event.value/1000000
         }
-        attrData.each {
-                def map = [:]
-                if (it.clusterInt == 0x0702 && it.attrInt == 0x0400) {
-                        log.debug "meter"
-                        map.name = "power"
-                        map.value = zigbee.convertHexToInt(it.value)/1000
-                }
-                if (it.clusterInt == 0x0702 && it.attrInt == 0x0000) {
-                         log.debug "energy"
-                         map.name = "energy"
-                         map.value = zigbee.convertHexToInt(it.value)/1000000
-                }
-                
-                if (map) {
-                        result << createEvent(map)
-                }
-                log.debug "Parse returned $map"
-        }
-        return result      
-    }
+        log.info "event outer:$event"
+		sendEvent(event)
+	} else {
+		List result = []
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		log.debug "Desc Map: $descMap"
+
+		List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
+		descMap.additionalAttrs.each {
+			attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+		}
+		attrData.each {
+				def map = [:]
+
+				if (it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_HISTORICAL_CONSUMPTION) {
+					log.debug "meter"
+					map.name = "power"
+					map.value = zigbee.convertHexToInt(it.value)/1000
```

Comment:
It just occurred to me, but it would be good to include the units, as well.
map.unit = "W"